lib/db.js

```diff
@@ -654,10 +667,13 @@ module.exports = (
     }
     return P.all(promises)
       .spread((deleteRes, redisTokens) => {
-        const redisSessionTokens = redisTokens ? JSON.parse(redisTokens) : []
-        const tokenToDeleteIndex = redisSessionTokens.findIndex((tok) => tok.tokenId === sessionToken.id)
-        redisSessionTokens.splice(tokenToDeleteIndex, 1)
-        return this.redis.setAsync(uid, JSON.stringify(redisSessionTokens))
+        if (this.redis) {
```

Comment:
Fwiw, for testing this, `test/local/db.js` already contains an example of a test that injects a mock redis object. All we need is a few little tests that return `null` or `undefined` from `createClient` instead, with some assertions that these db methods don't reject / return something sensible. Should be pretty easy, unless I'm overlooking something?